src/components/common/text/FiatText.js

```diff
@@ -11,17 +14,26 @@ import { useFiatText } from '../../../hooks/useFiatText'
 export type FiatTextFormat = 'raw' | 'primary' | 'secondary'
 
 type Props = {
-  format: FiatTextFormat,
+  walletId: string,
+  tokenId?: string,
   nativeCryptoAmount: string,
-  cryptoCurrencyCode: string,
-  cryptoExchangeMultiplier: string,
-  isoFiatCurrencyCode: string
+  format: FiatTextFormat
 }
 
-export const FiatText = ({ format, nativeCryptoAmount, cryptoCurrencyCode, cryptoExchangeMultiplier, isoFiatCurrencyCode }: Props) => {
+/**
+ * Return a formatted fiat text string representing the exchange rate of a
+ * specific crypto asset and native amount.
+ **/
+export const FiatText = ({ walletId, tokenId, nativeCryptoAmount, format }: Props) => {
+  const fiatCurrencyCode = useSelector(state => state.core.account.currencyWallets[walletId].fiatCurrencyCode)
```

Comment:
Same thing here with the incorrect `useSelector` usage. Just grab the account, then access everything else from that using normal non-hook code.